### ISSUE TITLE
[feature request] improved support for unicode literals

### ISSUE THREAD

**Issue by alecjacobson** [pressureless/linear_algebra#40]
```I❤️LA
10⁻¹²
```
should compile the same as 
```I❤️LA
10^(-12)
```
or
```I❤️LA
1E-12
```

**Comment by alecjacobson** [comment]
Similarly,

```I❤️LA
A : ℝⁿ```

should compile as


```I❤️LA
A : ℝ^(n)```

**Comment by alecjacobson** [comment]
This starts to get a bit esoteric. I think we _could_ but it seems an order of magnitude less important than unary minus operator.

**Comment by yig** [comment]
Yes, superscript arithmetic is a step too far. Really, long unicode superscripts and subscripts seem like a formatting issue to me. I can't imagine anyone will type more than a single letter. For me, for one letter, `^2` is replaced with `²`. Leaving it as `^2` doesn't change the number of characters I type. To type `²³` I need to type `^2^3`, so double the symbols. This doesn't scale well. It still wins for two characters, since the alternative is five characters long: `^(23)`. It loses for anything longer.

**Comment by alecjacobson** [comment]
This is a somewhat dubious use of `ˣ` but we could make our cartesian product declarations very unicode-beautiful with support for things like:

```
A ∈ ℝⁿˣ³
```

**Comment by yig** [comment]
Is that a superscript x or a superscript ×? I don't think I like this.

**Comment by alecjacobson** [comment]
X unfortunately
